### ISSUE TITLE
Package Manager UI Network errors should display in VS Errors List

### ISSUE THREAD

**Issue by jzabroski** [NuGet/Home#13884]
### NuGet Product(s) Affected

Visual Studio Package Management UI

### Current Behavior

In the last year, the behavior of this was improved to display errors in the InfiniteScrollList control that displays the "Manage Packages for Solution" window 

### Desired Behavior

Network errors during restore should display in Visual Studio Errors List, or at least the Package Manager Console output.

### Additional Context

I selected "Visual Studio Package Management UI" for this, but as I noticed, ideally this would belong to the VS Errors List component.

https://github.com/NuGet/NuGet.Client/pull/5286#issuecomment-2438614850

**Comment by donnie-msft** [comment]
I agree with the idea of logging these occurrences. What do you think about the Output window instead of the Error List? Its best suited for logging, whereas the Error List is more for error coded entries.

**Comment by jzabroski** [comment]
I was thinking that it would imply a new error code entry, but that may be path dependent. e.g., if this truly only happens through the UI in visual studio and cannot happen elsewhere in the nuget ecosystem like on the command line, then it probably does not deserve an error code.

I suppose, zooming out, the Output window _should_ report restore failure if there was a partial failure somewhere in the chain. So, probably the first deliverable is a failing unit test demonstrating partial failures are not handled correctly.

I doubt Microsoft standards would allow it since I never see Microsoft use third party APIs in unit tests, but typically I use Richard Szalay's MockHttpClient nuget package to mock these sorts of failures.  https://github.com/richardszalay/mockhttp
